nautilus-extension/nautilus-gsconnect.py

```diff
@@ -2,85 +2,75 @@
 nautilus-gsconnect.py - A Nautilus extension for sending files via GSConnect.
 
 A great deal of credit and appreciation is owed to the indicator-kdeconnect
-developers for the sister Python script 'kdeconnect-send-nautilus.py':
+developers for the sister Python script "kdeconnect-send-nautilus.py":
 
 https://github.com/Bajoja/indicator-kdeconnect/blob/master/data/extensions/kdeconnect-send-nautilus.py
 """
 
-import gettext
-import os.path
-import sys
+from gettext import translation
+from os.path import join, exists
+from sys import argv
 
 import gi
-gi.require_version('Gio', '2.0')
-gi.require_version('GLib', '2.0')
-gi.require_version('GObject', '2.0')
-from gi.repository import Gio, GLib, GObject
+gi.require_version("Gio", "2.0")
+gi.require_version("GLib", "2.0")
+gi.require_version("GObject", "2.0")
+from gi.repository.Gio import (
+    DBusProxy, DBusProxyFlags, BusType, DBusActionGroup, DBusCallFlags)
+from gi.repository.GLib import get_user_data_dir, Variant
+from gi.repository.GObject import Object
```

Comment:
Given that Python has an `object` builtin, this strikes me as a colossally terrible idea. "`Object`" alone does nothing to help future code readers understand that the object being discussed is a `GObject.Object`. Ditto `GLib.Variant` on the previous line.

Reply:
You don't need to know what module a object or function comes from to know what it does.  And to check you can quickly copy the import header then

Help(function_name) to find everything you need to know about it.

Any good IDE will know what your doing as well

Reply:
> You don't need to know what module a object or function comes from to know what it does.

Consider that a bit more, and I think you'll find that it's actually _**impossible**_ to know what an object or function does, unless you know its definition -- which requires knowing what module it's from.

Some IDEs can keep track of that for you, sure. Just like the interpreter keeps track of it in the code itself. But none of that changes the fact that **something** has to hold on to that information. If you don't know exactly what class is represented by `Object`, you know _nothing_ about it.

And when reading code in a non-interactive environment (like, say, a GitHub PR diff), there is no IDE hovering over your shoulder to remind you that `Object` is a `GObject.Object`.